### PR TITLE
Modify patches and update sub module

### PR DESCRIPTION
**Pull request by vasireddy99**
**Description:** 

Update Otel lambda submodule
- collector version patch
-  modify java patch file to not include any new changes form the upstream

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go/build.sh [hunk 1]
```diff
@@ -9,4 +9,5 @@ popd || exit
 # Build sample app
 
 cd ../opentelemetry-lambda/go/sample-apps/function || exit
+go get github.com/golang/protobuf@latest
```
Comment (bryan-aguilar) [inline go/build.sh:12]: Why is this necessary now?

Comment (vasireddy99) [reply go/build.sh:12]: Doing this resolves the[ protobuf failure](https://github.com/aws-observability/aws-otel-lambda/actions/runs/9020406554/job/24785526439#step:12:57) in workflow - https://github.com/protocolbuffers/protobuf/issues/16163,

Comment (vasireddy99) [reply go/build.sh]: updated the PR to include sub module changes, instead of modifying the build script.

**Review by bryan-aguilar — APPROVED** [review]
